### PR TITLE
feat(data-exploration): person and cohort tables

### PR DESCRIPTION
**Pull request by mariusandra**
## Problem

Follow up to https://github.com/PostHog/posthog/pull/13116

## Changes

Replaces persons and cohort tables with data exploration versions.

I had to do some annoying refactors to get the delete button properly extracted, apologies for all the noise around this in the PR:

I also cleaned up the storybook query stories somewhat.

## How did you test this code?

Clicking around in the interface. Hopefully soon with visual regression tests.

### CODE REVIEW

**Comment by mariusandra** [comment]
Actually ready for a look!

File: frontend/src/scenes/persons/PersonsScene.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+export function PersonsScene(): JSX.Element {
+    const { featureFlags } = useValues(featureFlagLogic)
+    const featureDataExploration = featureFlags[FEATURE_FLAGS.DATA_EXPLORATION_LIVE_EVENTS]
+    const { query } = useValues(personsSceneLogic)
+    const { setQuery } = useActions(personsSceneLogic)
+
+    return (
+        <>
+            <PersonPageHeader />
```
Comment (thmsobrmlr) [inline frontend/src/scenes/persons/PersonsScene.tsx:22]: Can we remove this header from `Persons`?

File: frontend/src/scenes/persons/PersonsScene.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import { useActions, useValues } from 'kea'
+import { featureFlagLogic } from 'lib/logic/featureFlagLogic'
+import { FEATURE_FLAGS } from 'lib/constants'
+import { Query } from '~/queries/Query/Query'
+import { Persons } from 'scenes/persons/Persons'
+import { PersonPageHeader } from 'scenes/persons/PersonPageHeader'
+import { personsSceneLogic } from 'scenes/persons/personsSceneLogic'
+
+export const scene: SceneExport = {
+    component: PersonsScene,
```
Comment (thmsobrmlr) [inline frontend/src/scenes/persons/PersonsScene.tsx:11]: Can we remove the scene export from `Persons`?

**Review by thmsobrmlr — CHANGES_REQUESTED** [review]
Great! Only one real issue that I ran into: The delete button for persons in cohorts does not show up for me.

Besides that some UX details regarding exports on the persons page, and a few inline comments. Nothing that needs to change, just observations:
- The exported CSV for "selected columns" contains a `person.$delete` column, which is probably unnecessary.
- The exported CSV for "all columns" contains a `value_at_data_point` column, which is probably unnecessary.
- Not related to this PR, but we might be able to improve this with data exploration - The exported CSV contains columns `id` and `distinct_ids.0`, but I can't filter for these columns (unclear what property I should use).
- The export button does not reflect the number of selected columns this any more:
    
    Before
    Screenshot 2022-12-08 at 12 00 58

    After
    Screenshot 2022-12-08 at 12 00 54

File: frontend/src/scenes/cohorts/CohortEdit.tsx [hunk 3]
```diff
@@ -193,7 +198,7 @@ export function CohortEdit({ id }: CohortLogicProps): JSX.Element {
                     </>
                 )}
 
-                {!isNewCohort && (
+                {typeof cohort.id === 'number' && (
```
[the remark this answers is not in the document]
Comment (mariusandra) [reply frontend/src/scenes/cohorts/CohortEdit.tsx:202]: Unfortunately this is needed for typing below when passing the cohort id to the query :/

**Comment by mariusandra** [comment]
Thanks for the review! Fixed all the feedback, I hope. Would love to get this in to help move along work on the two [stacked](https://github.com/PostHog/posthog/pull/13203) [PRs](https://github.com/PostHog/posthog/pull/13233) on top of this, plus the [event table regressions](https://github.com/PostHog/posthog/issues/13223)... 

> * The exported CSV for "selected columns" contains a `person.$delete` column, which is probably unnecessary.

Removed!

> * The exported CSV for "all columns" contains a `value_at_data_point` column, which is probably unnecessary.

This is there also on the current live persons table export. I propose we deal with it later 🙃 

> * Not related to this PR, but we might be able to improve this with data exploration - The exported CSV contains columns `id` and `distinct_ids.0`, but I can't filter for these columns (unclear what property I should use).

Hopefully HogQL will help here

> * The export button does not reflect the number of selected columns this any more:

Fixed!

File: frontend/src/scenes/cohorts/CohortEdit.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (thmsobrmlr) [reply frontend/src/scenes/cohorts/CohortEdit.tsx:202]: I think a comment would be great in that case :)

**Review by thmsobrmlr — APPROVED** [review]
🚢  it!

**Pull request merged**